src/middlewares/routeActivation.ts

```diff
@@ -106,4 +106,19 @@ export class RouteActivationMiddleware {
       RouteActivationMiddleware.shouldActivateRoute(destination, deliveryFilterList),
     );
   }
+
+  // This middleware will be used by source endpoint when we completely deprecate v0, v1 versions.
+  public static isVersionAllowed(ctx: Context, next: Next) {
+    const { version } = ctx.params;
+    if (version === 'v0' || version === 'v1') {
+      ctx.status = 500;
+      ctx.body =
+        '/v0, /v1 versioned endpoints are deprecated. Use /v2 version endpoint. This is probably caused because of source transformation call from an outdated rudder-server version. Please upgrade rudder-server to a minimum of 1.xx.xx version.';
```

Comment:
better to wrap in JSON structure

Reply:
This is not used as of now. Getting rid of this code to keep it clean.